pkg/sql/opt/props/func_dep.go

```diff
@@ -551,6 +551,23 @@ func (f *FuncDepSet) CopyFrom(fdset *FuncDepSet) {
 	f.hasKey = fdset.hasKey
 }
 
+// RemapFrom copies the given FD into this FD, remapping column IDs according to
+// the given function.
+func (f *FuncDepSet) RemapFrom(fdset *FuncDepSet, remap func(opt.ColumnID) opt.ColumnID) {
```

Comment:
There's a function `TranslateColSet` in opt/colset.go that already has this sort of logic - you could just pass in the left and output ColLists here and call `TranslateColSet`